### PR TITLE
UCT/IB: Fix subnet prefix for local IPoIB address

### PR DESCRIPTION
**Pull request by yosefe**
When the address is IB-local, qp_attr.ah_attr.grh.dgid is not
initialized, so we should take subnet prefix from loca gid[0]. Interface
id can be left unspecified, to make sure it's unpacked as 0 and not used
for address handle creation.

### CODE REVIEW

**Comment by yosefe** [comment]
@evgeny-leksikov pls take a look

File: src/uct/ib/rdmacm/rdmacm_cm.c [hunk 1]
```diff
@@ -171,28 +174,39 @@ static ucs_status_t uct_rdmacm_cm_id_to_dev_addr(struct rdma_cm_id *cm_id,
...
+    } else if (qp_attr.ah_attr.is_global) {
+        gid                = qp_attr.ah_attr.grh.dgid;
+        address_pack_flags = UCT_IB_ADDRESS_PACK_FLAG_SUBNET_PREFIX |
+                             UCT_IB_ADDRESS_PACK_FLAG_INTERFACE_ID;
     } else {
-        address_pack_flags = UCT_IB_ADDRESS_PACK_FLAG_INTERFACE_ID |
-                             UCT_IB_ADDRESS_PACK_FLAG_SUBNET_PREFIX;
+        /* For local IB address, assume the remote subnet prefix is the same
+         * and pack it to make reachability check pass*/
+        ret = ibv_query_gid(cm_id->verbs, cm_id->port_num, 0, &gid);
```
Comment (evgeny-leksikov) [inline src/uct/ib/rdmacm/rdmacm_cm.c]: maybe `UCT_IB_MD_DEFAULT_GID_INDEX`?

Comment (yosefe) [reply src/uct/ib/rdmacm/rdmacm_cm.c]: @evgeny-leksikov fixed

File: src/uct/ib/rdmacm/rdmacm_cm.c [hunk 2]
```diff
@@ -171,28 +174,40 @@ static ucs_status_t uct_rdmacm_cm_id_to_dev_addr(struct rdma_cm_id *cm_id,
...
+                             UCT_IB_ADDRESS_PACK_FLAG_INTERFACE_ID;
     } else {
-        address_pack_flags = UCT_IB_ADDRESS_PACK_FLAG_INTERFACE_ID |
-                             UCT_IB_ADDRESS_PACK_FLAG_SUBNET_PREFIX;
+        /* For local IB address, assume the remote subnet prefix is the same
+         * and pack it to make reachability check pass*/
+        ret = ibv_query_gid(cm_id->verbs, cm_id->port_num,
+                            UCT_IB_MD_DEFAULT_GID_INDEX, &gid);
+        if (ret) {
+            ucs_error("ibv_query_gid(dev=%s port=%d index=0) failed: %m",
```
Comment (evgeny-leksikov) [inline src/uct/ib/rdmacm/rdmacm_cm.c]: `... index=%d ... UCT_IB_MD_DEFAULT_GID_INDEX`

**Comment by yosefe** [comment]
@evgeny-leksikov fixed

**Review by evgeny-leksikov — APPROVED** [review]

**Pull request merged**